### PR TITLE
[WIP] Improve test coverage

### PR DESCRIPTION
**Pull request by luizvarela**
This PR tries to improve the test coverage described in https://github.com/devcenter-square/disease-info/issues/42, the initial idea is to write more tests to the Scraper class and then we can refactor, there are a lot of patterns that can be used in there.

For now the `Scrapers::WhoScraper`  class only have one spec, I will keep working on it.

the actual coverage is now:
coverage
but this number didn't say anything, we still need more specs to cover the edge cases.
- [x] Refactor disease scraper
- [x] Improve scraper tests
- [x] Improve disease model tests
- [x] Improve disease controller tests
- [x] Improve tests structure

### CODE REVIEW

**Comment by x6iae** [comment]
My first comment is this: I'm just waking up... and getting greeted with this PR. Even without looking yet, it's already made me a happy man. Today is a good day!!! 😁

File: spec/support/webmock.rb [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-      with(:headers => {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent'=>'ruby'}).
-      to_return(:status => 200, :body => body_mock, :headers => {})
-  end
-end
\ No newline at end of file
+# def body_mock
+#   ["disease"]
+# end
+#
+# RSpec.configure do |config|
```
Comment (x6iae) [inline spec/support/webmock.rb]: Are you still planning to use this? if it has expired it's lifespan, I think we can get rid of the whole file.

Comment (luizvarela) [reply spec/support/webmock.rb]: We can remove it if I need to put some webmock config I can create it again. I just commented out to disable this call.

Comment (x6iae) [reply spec/support/webmock.rb]: Yes please.

**Comment by luizvarela** [comment]
Almost there, still working on the last specs.
new-coverage

**Comment by x6iae** [comment]
Go @luizvarela, Go @luizvarela, Go @luizvarela... 👍 👍 👍

Will wait for when you are through with the last specs.
You rock, keep rocking. 🚀 🚀 🚀

**Comment by x6iae** [comment]
@luizvarela Could you also check for the conflicts so this will be mergeable?

**Comment by luizvarela** [comment]
Now the test coverage is 100%. :champagne: :tada: 

I did not remove the original parser, `who_scraper` yet. If everything is ok I can delete it.



In this PR I didn't change any parser logic, I`ve just applied some code refactor and improve the code design. We can still improve the parser in different pull requests. Feel free to send suggestions.

**Comment by AfolabiOlaoluwa** [comment]
Well done. Really nice to have you on this project.

**Comment by AfolabiOlaoluwa** [comment]
I agree to remove original parser `who_srapper`. leaving it there makes the repo DRY. From the folder/directory called who, I see the parser logic didn't change. 
I concur to merge.

**Comment by luizvarela** [comment]
Thanks @SundayAdefila @AfolabiOlaoluwa @andela-rsadiq for review and accept this pull request. :tada:

**Comment by x6iae** [comment]
Cool!!... 

I guess we can proceed to remove the original `who_scrapper` now... prolly in another PR... then get this baby into the mainstream of things!
